pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/SimpleFilter.java

```diff
@@ -49,22 +54,30 @@
  *     @Property(name="service.ranking", intValue=1)
  * })
  */
-//@SlingFilter(order=1, description="A Simple Filter")
-//@Property(name="service.vendor", value="The Apache Software Foundation")
+@Component
+@SlingServletFilter(scope = {SlingServletFilterScope.REQUEST},
+                    methods = {"GET","HEAD"})
 public class SimpleFilter implements Filter {
     
+    public static final String DOMAIN_ALLOWED = ".redhat.com";
     private final Logger log = LoggerFactory.getLogger(SimpleFilter.class);
 
     public void init(FilterConfig filterConfig) throws ServletException {
     }
 
-    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException,
+    public void doFilter(final ServletRequest req, final ServletResponse res, final FilterChain chain) throws IOException,
             ServletException {
-        log.info("filter invoked - start");
+        final SlingHttpServletRequest request = (SlingHttpServletRequest)req;
+        final SlingHttpServletResponse response = (SlingHttpServletResponse)res;
+        String origin = request.getHeader("Origin");
+        if (origin != null) {
+            if (origin.contains(DOMAIN_ALLOWED)) {
```

Comment:
not a blocker, but since this is a security-related change, I would make this condition a bit tighter. Maybe `endsWith` would work, or a regular expression check. 
Maybe I'm being too careful here.. so I'll leave this decision to you.

Reply:
updated